incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -207,7 +241,8 @@ private MqttSubscribeProxy(
             this.routedId = routedId;
             this.initialId = initialId;
             this.replyId = supplyReplyId.applyAsLong(initialId);
-            this.delegate = new KafkaProxy(originId, resolvedId, this);
+            this.messages = new KafkaMessagesProxy(originId, resolvedId, this);
+            this.retained = new KafkaRetainedProxy(originId, resolvedId, this);
```

Comment:
The `resolvedId` could be different for `messages` vs `retained`, based on routing, right?

Suggest using the information from the `KafkaBeginEx` in the factory newStream method to determine if `retained` is needed, passing `0` or `non-zero` for `retainedId` (resolved retained id).

Then retained would be initialized only for non-zero `retainedId`, and subsequent methods would test for `retained != null` instead of `retainAvailable`, which can then be removed from `MqttKafkaConfiguration`.

Reply:
"The resolvedId could be different for messages vs retained, based on routing, right?"
I think we didn't talk about this yet. Do we want to allow the user to control the mqtt_retained details? Let's talk about this.